### ISSUE TITLE
Error occurs when calling torch.save() in Python2.7

### ISSUE THREAD

**Issue by Response777** [pytorch/pytorch#819]
My codes break when calling `torch.save()` with `AttributeError: 'unicode' object has no attribute 'tell'`

Here are the error logs.

```
print(model)
```

```
RNN (
  (i2h): Linear (214 -> 128)
  (i2o): Linear (214 -> 19)
  (softmax): Softmax ()
)
```

```
torch.save(obj = model, f ='./models/RNN.pth')
```

```
AttributeErrorTraceback (most recent call last)
<ipython-input-32-5838882c81ba> in <module>()
----> 1 torch.save(obj = model, f ='./models/RNN.pth')

/home/response777/.local/lib/python2.7/site-packages/torch/serialization.pyc in save(obj, f, pickle_module, pickle_protocol)
    121         f = open(f, "wb")
    122     try:
--> 123         return _save(obj, f, pickle_module, pickle_protocol)
    124     finally:
    125         if new_fd:

/home/response777/.local/lib/python2.7/site-packages/torch/serialization.pyc in _save(obj, f, pickle_module, pickle_protocol)
    212         pickle_module.dump(sys_info, f, protocol=pickle_protocol)
    213 
--> 214     with closing(tarfile.open(fileobj=f, mode='w:', format=tarfile.PAX_FORMAT)) as tar:
    215         _add_to_tar(save_sys_info, tar, 'sys_info')
    216         _add_to_tar(pickle_objects, tar, 'pickle')

/usr/lib/python2.7/tarfile.pyc in open(cls, name, mode, fileobj, bufsize, **kwargs)
   1676             else:
   1677                 raise CompressionError("unknown compression type %r" % comptype)
-> 1678             return func(name, filemode, fileobj, **kwargs)
   1679 
   1680         elif "|" in mode:

/usr/lib/python2.7/tarfile.pyc in taropen(cls, name, mode, fileobj, **kwargs)
   1703         if mode not in ("r", "a", "w"):
   1704             raise ValueError("mode must be 'r', 'a' or 'w'")
-> 1705         return cls(name, mode, fileobj, **kwargs)
   1706 
   1707     @classmethod

/usr/lib/python2.7/tarfile.pyc in __init__(self, name, mode, fileobj, format, tarinfo, dereference, ignore_zeros, encoding, errors, pax_headers, debug, errorlevel)
   1564         self.members = []       # list of members as TarInfo objects
   1565         self._loaded = False    # flag if all members have been read
-> 1566         self.offset = self.fileobj.tell()
   1567                                 # current position in the archive file
   1568         self.inodes = {}        # dictionary caching the inodes of

AttributeError: 'unicode' object has no attribute 'tell'
```

**Comment by Response777** [comment]
After check, I found that this is resulted by `from __future__ import unicode_literals`.

It seems that the authors forget to deal with situation where the filename is actually `unicode` in python2.7.

**Comment by soumith** [comment]
thanks for the investigation. yes, i think we dont properly deal with unicode filenames in py2.
We'll fix this.

**Comment by Response777** [comment]
Just in case, the `torch.load` share the exactly same problem.

**Comment by apaszke** [comment]
This is being fixed in #713.

**Comment by soumith** [comment]
fixed now in master
